### PR TITLE
x/ref/runtime/internal/rpc,v23/requestid: add requestid to flowserver…

### PR DESCRIPTION
**Pull request by noahtseng**
… ctx

### CODE REVIEW

**Comment by cosnicolaou** [comment]
See #154 - if you move this package to x/ref/runtime/internal/requestid then you can use the context.WithLoggingPrefix to set this value and the logging 'should just work' from there.

File: v23/requestid/model.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	"v.io/v23/context"
+)
+
+type contextKey int
+
+const requestIDKey = contextKey(iota)
+
+func WithNewRequestID(ctx *context.T) *context.T {
+	requestID := uuid.New()
+	return  context.WithValue(ctx, requestIDKey, requestID)
```
Comment (cosnicolaou) [inline v23/requestid/model.go:14]: with #154, you can just use return context.WithLoggingPrefix(ctx, requestID).

Comment (noahtseng) [reply v23/requestid/model.go:14]: Should we still include this so we support different loggers? 

For example, we may want to use a logger that logs in json format.

Comment (noahtseng) [reply v23/requestid/model.go:14]: I think we can both:
```
func WithNewRequestIDLoggingPrefix(ctx *context.T) *context.T {
	requestID, err := uuid.NewUUID()
	if err != nil {
		panic(err)
	}
	return  context.WithLoggingPrefix(ctx, requestID)
}
func WithNewRequestID(ctx *context.T) *context.T {
	requestID, err := uuid.NewUUID()
	if err != nil {
		panic(err)
	}
	return  context.WithValue(ctx, requestIDKey, requestID)
}
```

Comment (cosnicolaou) [reply v23/requestid/model.go]: yes, can certainly have both if you think there's a need for both, but the way you have the API setup now there would always be two different request IDs, so I'm curious to know what your use case for that it is? If you want to keep the WithNewRequestID then I'd suggest not having WithNewRequestIDLoggingPrefix and instead do something like:


```go
func WithNewRequestID(ctx *context.T) (*context.T, ID) {
	requestID, err := uuid.NewUUID()
	if err != nil {
		panic(err)
	}
	return  context.WithValue(ctx, requestIDKey, requestID), requestID
}

ctx, id = requestid.WithNewRequestID(ctx)
ctx = context.WithLoggingPrefix(ctx, id)
```

Or if you don't want to return the ID from `WithNewRequestID`
```go
ctx = requestid.WithNewRequestID(ctx)
ctx = context.WithLoggingPrefix(ctx, requestid.RequestID(ctx)) // get the existing request ID from the context.
```

Comment (noahtseng) [reply v23/requestid/model.go]: The use case I was considering is for a custom logger (e.g. json formatted logging). But, I now see that we just prepend the `requestid` to the list of `args` in `prefixedArgs()`, which should allow the logger to handle the args however it wants (including formatting). 

In that case, we shouldn't need `WithNewRequestID()` as you suggested.

Comment (cosnicolaou) [reply v23/requestid/model.go]: yes, that's correct. The custom logger would be implemented by replacing x/ref/internal/logger with a different package that implements your custom logger.

**Comment by noahtseng** [comment]
This PR should be ready for review whenever you get the chance!

**Review by cosnicolaou — APPROVED** [review]
looks good to me, thanks!

**Pull request closed without merging**